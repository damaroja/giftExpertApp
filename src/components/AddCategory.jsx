



























import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
  const [InputValue, setInputValue] = useState('');

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if(InputValue.trim().length <= 1) return
    onNewCategory(InputValue.trim())
    setInputValue('')
    
  };

  return (
    <form onSubmit={(e) => onSubmit(e)} aria-label="form">
      <input
        type="text"
        value={InputValue}
        placeholder="Buscar Gifts"
        onChange={(e) => onInputChange(e)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
