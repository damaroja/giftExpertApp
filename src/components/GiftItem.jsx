


























import PropTypes from 'prop-types'


export const GiftItem = ({title, url}) => {
  return (
    <>
    <div className="card">
          <p>{title}</p>
          <p><img src={url} alt={title}/></p>
          </div>
    </>
  )
}


GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired
}
