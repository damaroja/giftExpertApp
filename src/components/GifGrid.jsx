






















import React from "react";
import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftItem } from "./GiftItem";
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {
  
  const { Images, isLoading } = useFetchGifts(category)

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<h2>Cargando...</h2>)
      }
      <div className="card-grid">
        {Images.map((image) => {
          return <GiftItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
