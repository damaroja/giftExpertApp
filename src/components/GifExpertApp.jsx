import React from "react";
import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";
































export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp2</h1>
      <AddCategory onNewCategory={(event) => onAddCategory(event)} />
      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
