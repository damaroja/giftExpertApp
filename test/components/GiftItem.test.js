











/* export const GiftItem = ({title, url}) => {
    return (
      <>
      <div className="card">
            <p>{title}</p>
            <p><img src={url} /></p>
            </div>
      </>
    )
  }
  
  
  GiftItem.Proptypes = {
    title: Proptypes.string.isRequired,
    url  : Proptypes.string.isRequired
  } */

import { GiftItem } from "../../src/components/GiftItem";
import { render, screen } from "@testing-library/react";

describe("Test del <GiftItem />", () => {
  const titulo = "ESte es el titulo";
  const address = "https://media0.giphy.com/media/jhkgdsf7683gfd763";
  test("Test del snapshot ", () => {
    const { container } = render(<GiftItem title={titulo} url={address} />);
    expect(container).toMatchSnapshot();
  });
  test("Debe de mostrar la imagen con el alt y el URL indicado", () => {
    render(<GiftItem title={titulo} url={address} />);
    const { src, alt } = screen.getByRole("img");
    expect(alt).toBe(titulo);
    expect(src).toBe(address);
});
test('Test para mostrar el titulo del componente', () => {
    render(<GiftItem title={titulo} url={address} />);
    expect(screen.getByText(titulo)).toBeTruthy();
  });
});
