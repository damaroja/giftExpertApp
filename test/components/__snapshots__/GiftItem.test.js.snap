// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del <GiftItem /> Test del snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <p>
      ESte es el titulo
    </p>
    <p>
      <img
        alt="ESte es el titulo"
        src="https://media0.giphy.com/media/jhkgdsf7683gfd763"
      />
    </p>
  </div>
</div>
`;
